File: app-mvvm/src/main/java/com/xuie/imaginary/util/DateUtils.java
package xuk.imaginary.common;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * @author xuie
 * @date 17-8-15
 */
public class DateUtils {
    private static final String MATCH_REGEX = "(\\d+)-(\\d+)-(\\d+)T(\\S+)Z";

    public static String getDate(String s) {
        Matcher m = Pattern.compile(MATCH_REGEX).matcher(s);
        if (!m.matches() || m.groupCount() < 4) {
            return "";
        }
        return m.group(1) + "/" + m.group(2) + "/" + m.group(3);
    }
}

File: app-mvvm-kodein/src/main/kotlin/com/xuie/imaginary/util/DateUtils.kt
package com.xuie.imaginary.util

import java.util.regex.Pattern

/**
 * @author Jie Xu
 * @date 17-8-15
 */
object DateUtils {
  private const val MATCH_REGEX = "(\\d+)-(\\d+)-(\\d+)T(\\S+)Z"

  fun getDate(s: String?): String {
    if (s.isNullOrEmpty()) {
      return ""
    }
    val m = Pattern.compile(MATCH_REGEX).matcher(s)
    return if (m == null || !m.matches() || m.groupCount() < 4) {
      ""
    } else m.group(1) + "/" + m.group(2) + "/" + m.group(3)
  }
}
